internal/cmd/cli/apply/apply.go

```diff
@@ -567,7 +571,11 @@ func saveOCIBundle(ctx context.Context, c client.Client, r record.EventRecorder,
 
 	updated := bundle.DeepCopy()
 	_, err = controllerutil.CreateOrUpdate(ctx, c, bundle, func() error {
-		if bundle != nil && bundle.Spec.HelmOpOptions != nil {
+		if bundle.DeletionTimestamp != nil {
+			return fmt.Errorf("the bundle %q is being deleted", bundle.Name)
+		}
```

Comment:
This only covers the case where the Bundle is marked for deletion during the execution of the function (`DeletionTimestamp` not set before calling `CreateOrUpdate`). Assuming that `bundle` passed to `saveOCIBundle` is "produced" from scratch with the desired state, this still has the risk of doing extra work unnecessarily (`pushOCIManifest`) for a bundle that is already being deleted and will never be used. Should we do an early `get` of the `Bundle` to prevent that?